File: dart/test/number_validator_test.dart

import '../lib/src/v1/utils/number_validator.dart';
import 'package:test/test.dart';

void main() {
  test('validateNumbers', () {
    final numbersToValidate = [
      '256712345678',
      '+256712345678',
      '0712345678',
      '235-787-900-123',
      '+257 700 567 234',
      '0745',
    ];
    final validated = NumberValidator.validateNumbers(numbersToValidate);
    expect(validated, isNotNull);
    expect(validated.length, 3);
    expect(validated.contains('256712345678'), isTrue);
    print(validated);
  });
}

File: java/src/test/java/com/pahappa/systems/commssdk/v1/utils/NumberValidatorTest.java
package com.pahappa.systems.egosmssdk.v1.utils;

import org.junit.Test;

import java.util.Arrays;
import java.util.List;

import static org.junit.Assert.*;

public class NumberValidatorTest {

    @Test
    public void validateNumbers() {
        List<String> numbersToValidate = Arrays.asList(
                "256712345678", "+256712345678", "0712345678", "235-787-900-123",
                "+257 700 567 234", "0745"
        );
        List<String> validated = NumberValidator.validateNumbers(numbersToValidate);
        assertNotNull(validated);
        assertEquals(3, validated.size());
        assertTrue(validated.contains("256712345678"));
        System.out.println(validated);
    }
}